L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,369 @@
+  // on the server side, and optional for the client side(in the one-side
+  // TLS scenario, the client is not required to provide identity certs).
+  void watch_identity_key_cert_pairs();
+  // Sets the name of identity key-cert pairs being watched, if
+  // |watch_identity_key_cert_pairs| is called. If not set, an empty string will
+  // be used as the name.
+  //
+  // @param identity_cert_name the name of identity key-cert pairs being set.
+  void set_identity_cert_name(const std::string& identity_cert_name);
+  // Sets the Tls session key logging configuration. If not set, tls
+  // session key logging is disabled. Note that this should be used only for
+  // debugging purposes. It should never be used in a production environment
+  // due to security concerns.
+  //
+  // @param tls_session_key_log_file_path: Path where tls session keys would
+  // be logged.
+  void set_tls_session_key_log_file_path(
+      const std::string& tls_session_key_log_file_path);
+  // Sets the certificate verifier used to perform post-handshake peer identity
+  // checks.
+  void set_certificate_verifier(
+      std::shared_ptr<CertificateVerifier> certificate_verifier);
+
+  // Sets the crl provider, see CrlProvider for more details.
+  void set_crl_provider(std::shared_ptr<CrlProvider> crl_provider);
+
+  // Sets the minimum TLS version that will be negotiated during the TLS
+  // handshake. If not set, the underlying SSL library will use TLS v1.2.
+  // @param tls_version: The minimum TLS version.
+  void set_min_tls_version(grpc_tls_version tls_version);
+  // Sets the maximum TLS version that will be negotiated during the TLS
+  // handshake. If not set, the underlying SSL library will use TLS v1.3.
+  // @param tls_version: The maximum TLS version.
+  void set_max_tls_version(grpc_tls_version tls_version);
+
+  // Sets a custom chain builder implementation that replaces the default chain
+  // building from the underlying SSL library.
+  void set_custom_chain_builder(std::shared_ptr<CustomChainBuilder>
+  chain_builder);
+
+  // Configures if the client should verify the server cert (mTLS). Is a no-op
+  // if using these options to create a server credential.
+  void set_verify_server_cert(bool verify_server_cert);
+
+ // Sets the options of whether to request and/or verify client certs. This
+ // is a no-op if using these options to create a client credential.
+void set_cert_request_type(grpc_ssl_client_certificate_request_type type);
+```
+
+Here is an example of how to use this API:
+```c++
+/* Create option and set basic security primitives. */
+TlsCredentialsOptions options = TlsCredentialsOptions();
+options.set_verify_server_cert(true);
+options.set_min_tls_version(TLS1_2);
+options.set_max_tls_version(TLS1_3);
+
+// The core credentials APIs are still C, so this is still a grpc_channel_credential
+/* Create TLS channel credentials. */
+grpc_channel_credentials* creds = grpc_tls_credentials_create(options);
```

Comment:
This is not quite accurate, since `options` cannot be passed into `grpc_tls_credentials_create`.

Reply:
I think part of this would be changing that API to take `TlsCredentialsOptions` instead of `grpc_tls_credentials_options` (which won't exist anymore)
Or are you saying it should be a pointer instead of just the value?